velox/exec/tests/PartitionedOutputTest.cpp

```diff
@@ -0,0 +1,156 @@
+  getData(const std::string& taskId, int destination, int64_t sequence) {
+    auto [promise, semiFuture] = folly::makePromiseContract<
+        std::vector<std::unique_ptr<folly::IOBuf>>>();
+    VELOX_CHECK(bufferManager_->getData(
+        taskId,
+        destination,
+        PartitionedOutput::kMinDestinationSize,
+        sequence,
+        [result = std::make_shared<
+             folly::Promise<std::vector<std::unique_ptr<folly::IOBuf>>>>(
+             std::move(promise))](
+            std::vector<std::unique_ptr<folly::IOBuf>> pages,
+            int64_t /*inSequence*/,
+            std::vector<int64_t> /*remainingBytes*/) {
+          result->setValue(std::move(pages));
+        }));
+    auto future = std::move(semiFuture).via(executor_.get());
+    future.wait(std::chrono::seconds{10});
+    VELOX_CHECK(future.isReady());
+    return std::move(future).value();
+  }
+
+  std::vector<std::unique_ptr<folly::IOBuf>> getAllData(
+      const std::string& taskId,
+      int destination) {
+    std::vector<std::unique_ptr<folly::IOBuf>> result;
+    int attempts = 0;
+    bool done = false;
+    while (!done) {
+      attempts++;
+      VELOX_CHECK_LT(attempts, 100);
+      std::vector<std::unique_ptr<folly::IOBuf>> pages =
+          getData(taskId, destination, result.size());
+      for (auto& page : pages) {
+        if (page) {
+          result.push_back(std::move(page));
+        } else {
+          bufferManager_->deleteResults(taskId, destination);
+          done = true;
+          break;
+        }
+      }
+    }
+    return result;
+  }
+
+ private:
+  const std::shared_ptr<OutputBufferManager> bufferManager_{
+      OutputBufferManager::getInstance().lock()};
+};
+
+TEST_F(PartitionedOutputTest, flush) {
+  // This test verifies
+  //  - Flush thresholds are respected (flush doesn't happen neither too early
+  //  nor too late)
+  //  - Flush is done independently for each output partition (flush for one
+  //  partition doesn't trigger flush for another one)
+
+  auto input = makeRowVector(
+      {"p1", "v1"},
+      {makeFlatVector<int32_t>({0, 1}),
+       makeFlatVector<std::string>({
+           // twice as large to make sure it is always flushed (even if
+           // PartitionedOutput#setTargetSizePct rolls 120%)
+           generateString(PartitionedOutput::kMinDestinationSize * 2, '0'),
+           // 10 times smaller, so the data from 13 pages is always flushed as 2
+           // pages
+           // 130% > 120% (when PartitionedOutput#setTargetSizePct rolls 120%)
+           // 130% < 140% (when PartitionedOutput#setTargetSizePct rolls 70% two
+           // times in a row)
+           generateString(PartitionedOutput::kMinDestinationSize / 10, '1'),
+       })});
+
+  auto plan =
+      PlanBuilder()
+          // produce 13 pages
+          .values({input}, false, 13)
+          .partitionedOutput(
+              std::vector<std::string>{"p1"}, 2, std::vector<std::string>{"v1"})
```

Comment:
can you drop `std::vector<std::string>`?

Reply:
Dropped the first one. However when I try to drop the second one it picks up a wrong overload (`Type 'const char[3]' cannot be narrowed to 'bool' in initializer lis`)